src/main/java/org/owasp/esapi/SecurityConfiguration.java

```diff
@@ -640,6 +640,12 @@
      */
     InputStream getResourceStream( String filename ) throws IOException;
 
+    /**
+     * Used to load antisamy-esapi.xml from a variety of different classpath locations.
```

Comment:
This Javadoc is not sufficient.
First off, you have created a more general method than just something load an AntiSamy configuration file from the classpath. Instead this new method would allow any filename found in the class path to be retrieved. Is that a good thing? Maybe, but it is far beyond what is intended and I think we carefully need to consider any security implications.
Secondly, if we are going to leave this new method (still TBD in my opinion), then at a bare minimum, we need to carefully document the specific detailed order of the search for the `filename` parameter. Failure to do that will result on countless questions.

Reply:
Deserialization is something I need to understand a lot more.  Off the cuff, I agree with the private method assessment so we can control the behavior.

Reply:
Resolved by making this method private in HTMLValidationRule (even though it duplicates 90% of the code from DefaultSecurityConfiguration). But at least it removes the immediate security concerns that we had.